nntrainer/models/neuralnet.cpp

```diff
@@ -382,8 +382,9 @@ void NeuralNetwork::load(const std::string &file_path,
       /// this is assuming that the failure is allowed at the end of the file
       /// read. so, after this line, additional read shouldn't be called
       if (opt && istrequal(opt->getType(), "adam")) {
-        char opt_type[4];
-        model_file.read(opt_type, 4);
+        std::string opt_type;
```

Comment:
you might need to change the save function as well.

Reply:
Done